### ISSUE TITLE
Use Away instead of Offline for buddies

### ISSUE THREAD

**Issue by ildar** [ars3niy/tdlib-purple#80]
Buddies constantly falling into "Offline" status. This is unacceptable because:
1. They aren't offline, they are away from their phones. This is the big difference in UX in buddy list. (hide offline == true)
2. Pidgin rejects sending messages to those users.

I see that the change was intentional: https://github.com/ars3niy/tdlib-purple/commit/2d20954e6f2b66df9e3641c3832d1ce68de7cb2a

**Comment by ars3niy** [comment]
You would do well to be less categorical in your statements. Away does, however, make more sense than offline because it doesn't even mean they are away from the device I don't think – only that they aren't actively using telegram client right now.

It requires adjusting unit tests, though, so it's not quite a one-liner change.

**Comment by ildar** [comment]
Sorry for the wrong wording. Yet this is critical IMHO because
> 2. Pidgin rejects sending messages to those users

No chance i can use this plugin (and i think other pidgin users) until it's
fixed.  Thanks!

**Comment by ars3niy** [comment]
I don't understand it though, normally pidgin has no problem sending to offline buddies. I mean I'm using it with this plugin myself.

**Comment by ildar** [comment]
Hmm. I'll get you screenshots tomorrow then.
Still having buddies hidden as offline is frustrating.

**Comment by ildar** [comment]
Well, sending to the offline buddy results in this dialog.
Pressing NO sends the message ok. But overall this is a terrible UX.
Thanks for the understanding.

**Comment by The-Compiler** [comment]
FWIW I'd also prefer them being marked as away rather than offline. I'm using this plugin with Bitlbee, and contacts which are offline are not in the channel - so if I want to write something to them, I need to look up their nick manually somewhere.

**Comment by ars3niy** [comment]
Aha, it's because you're using that plugin.

Well anyway, Away was always a better choice, I'll change it.

**Comment by ildar** [comment]
thanks a lot!

**Comment by ars3niy** [comment]
Changed in latest version

**Comment by smirnov-d-a** [comment]
I have to regretfully disagree with this decision.
I, on the contrary, would like to see buddies as "Offline" instead of "Away", so they don't clutter up my buddylist, while they are not actually online in the current moment. Maybe there should be a timeout or whatever to make them go offline after some (adjustable) time, like 5 mins or something.

As for the above-mentioned problems with them becoming offline, I suppose they could (and should, actually) be solved by other means.
For example, if you want to see all your buddies in your buddylist, you can very well enable that in Pidgin's own preferences (_Buddies -> Show -> Offline Buddies_ menu), you can even adjust that individually for every buddy or group you need.
The same goes for offline messages. The described behaviour with the dialog appearing, while trying to send a message to offline contact, is more than likely triggered by the **_Offline Message Emulation_** plugin, which would do this on any protocol. So, that has nothing to do with the **_tdlib-purple_** plugin whatsoever.

Anyway, to summarize, I would propose to have this as an option - to show buddies as "Away" or "Offline", so everyone would have it as they like. Probably, there could even be an adjustable timeout for them to change status, but as for me, it's not that neccessary.
Actually, there was such an option in old _**telegram-purple**_ plugin, but unfortunately it never worked (see https://github.com/majn/telegram-purple/issues/379). So, when I found this new (and better still) plugin, I was going to make an similar issue, but, having found this discussion, I decided to post here for a start.
